### PR TITLE
Fixed an issue in ServerRepository, where clients can send incomplete…

### PR DESCRIPTION
**Pull request by SkippsDev**
… packets and crash the server.

### CODE REVIEW

File: direct/src/distributed/ServerRepository.py [hunk 1]
```diff
@@ -359,8 +359,11 @@ def handleDatagram(self, datagram):
             #datagram.dumpHex(ostream)
 
         dgi = DatagramIterator(datagram)
-
         type = dgi.getUint16()
+        
+        if dgi.getRemainingSize() == 0 and type != CLIENT_DISCONNECT_CMU:
+            # This prevents a crash where their isn't enough data recieved to unpack corresponding to the msgType.
```
Comment (mfwass) [inline direct/src/distributed/ServerRepository.py]: Might want to correct the grammar.

their -> there

**Comment by loblao** [comment]
I'd like to add that ServerRepository.py is not really "fixable", there are countless ways to exploit/crash it.

**Comment by SkippsDev** [comment]
There's one less way to crash it from now on.

**Comment by CFSworks** [comment]
While this fix is *functional* I don't think it's *correct*. It only fixes one manifestation of the "unexpected end of datagram" crash bug (a non-disconnect packet with no payload). What happens if you only send one byte of payload to something expecting multiple bytes? The check would allow the datagram through but it would still crash upon processing.

A better fix would be to ensure that all data is read from the dgi *first*, and wrap that in a try/except to catch reads past the end of the iterator. The best fix would be to have each message-type handler check the length before attempting to process the message. (And in either case, an error should probably result in the client getting dropped for violating the protocol.)

**Comment by SkippsDev** [comment]
However i was just graciously fixing an exploit because i noticed it was just something that needed to be done... I never indented to completely "fix" the server repository, just figured i would help out.

**Pull request closed without merging**